cmd/testgrid-config-generator/main.go

```diff
@@ -182,6 +183,94 @@ type release struct {
 	} `json:"verify"`
 }
 
+// e2ePresubmitJobsByBranch collects statically-defined e2e presubmit jobs (e2e,
+// serial and upgrade) grouped by the targeted branch.
+func e2ePresubmitJobsByBranch(presubmitBranches []string, jobConfig *prowconfig.JobConfig) map[string][]string {
+	presubmitJobs := map[string][]string{}
+	for _, p := range jobConfig.AllStaticPresubmits(nil) {
+		// Only track mandatory jobs
```

Comment:
This is not technically a Prow-ism, so if you were hoping for this to catch all that is required for merge, we need something else

Reply:
I care less about what is required to merge as limiting the scope of collection. There are a ton of optional on-demand jobs that wouldn't add much signal but would potentially slow testgrid collection down.

Reply:
Updated the comment to reflect my intent, does that make more sense?